ksqldb-rest-client/src/main/java/io/confluent/ksql/rest/client/KsqlRestClient.java

```diff
@@ -208,10 +208,10 @@ public void setServerAddress(final String serverAddress) {
   ) {
     KsqlTarget target = target();
     final Map<String, Object> requestPropertiesToSend = new HashMap<>(requestProperties);
-    if (ConsistencyOffsetVector.isConsistencyVectorEnabled(localProperties.toMap())) {
+    if (ConsistencyOffsetVector.isConsistencyVectorEnabled(requestPropertiesToSend)) {
       final String serializedCV = serializedConsistencyVector.get();
       // KsqlRequest:serializeClassValues throws NPE for null value
-      requestPropertiesToSend.put(
+      requestPropertiesToSend.putIfAbsent(
```

Comment:
If I make this change, `putIfAbsent` then I can test the behavior much easier. However, it is allowing someone write access to the consistency token. Then again, they have write access if they do a curl. @vvcephei @patrickstuedi 
 What do you think? Is it ok?

Reply:
I don't think it's a problem. But can you explain why using put makes testing more difficult?